x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -159,6 +166,9 @@ export class TaskRunner<
     }
   ) {
     const client = this.context.internalSavedObjectsRepository;
+    try {
+      await this.ruleRunning.waitFor();
```

Comment:
**question:** Where are we setting `running: false`? Should we move this code outside of `updateRuleSavedObject` and into the place that would call `updateRuleSavedObject` with a `running: false` flag so we don't always await this promise?

Reply:
let me think on this one, I did not think about that one. thank you!

Reply:
Ok I have my thought on this ones, I was thinking this function `updateRuleSavedObject` will only set the property `running` to be false (but I did forget to set it). Since, our running handler is taking care of setting the `running` field to be true.
Tell me if I am wrong here but we should not wait that much if the run is fast because the `waitFor` function will return `Promise.resolve` right away.